### ISSUE TITLE
VM Share with Team

### ISSUE THREAD

**Issue by gokmen** [koding/koding#10154]
We need to provide an option to share a VM with whole team not only with a given members.

To be able to do that we need to create another type of custom user entry in `JMachine.users` which will be used for **UI**, **Kontrol** and **Klient** if needed. Possible scenario can be like following;

 - User selects one of running vm to share with the team (target)
   1. we make a request to bongo api for adding `*` entry to `JMachine.users` list
   2. we make another request to target machine **Klient** to mark it as shared with team
 - This makes target machine to visible on all other users sidebar on **UI**
    (requires a change on `JMachine.some` selector to include machines which has `*` user)
 - A member of the team tries to connect to target vm
 - Requests comes along to target **Klient**
 - Since this machine is not explicitly shared with this member **Klient** asks to **Kontrol** if it should allow the user or not.
 - **Kontrol** responds `ok` if machine has `*` user, and the requester is member of the active team.
 - **Klient** adds this user into `boltdb` as a regular user of this machine and allows user to access machine as usual. (this part will prevent **Klient** to ask to **Kontrol** if this user tries to connect to this machine again)

**Comment by rjeczalik** [comment]
> - Since this machine is not explicitly shared with this member Klient asks to Kontrol if it should allow the user or not.
> - Kontrol responds ok if machine has * user, and the requester is member of the active team.

Should we use remote.api for this or query mongo directly? If yes, I'd add the api to kloud, rather than to kontrol.

**Comment by rjeczalik** [comment]
> But do we have a connection between klient to kloud?

Yes, we do now thanks to the new kites/config package. Klient and kd is able to connect directly to kloud, kontrol, remote.api and socialapi, depending on what's needed.
